[FEATURE] Ajoute la création d'une campagne à partir d'une campagne existante (PIX-8622).

HEYGUL opened the pull request:
## :unicorn: Problème

Nos prescripteurs ont énormément de campagnes à créer, pour lesquelles seuls quelques paramètres changent (comme le profil cible ou le titre de la campagne).
Au moment de la rentrée par exemple, les professeurs d'école de chaque classe créent des campagnes 6ième, 5ième, 4ième, ou encore les enseignants du supérieur pour toutes les différentes licences etc.

## :robot: Proposition

On permet le pré-remplissage du formulaire de création de campagne à partir des attributs d'une campagne source.
Ce pré-remplissage se fait en spécifiant l'identifiant de la campagne source dans un paramètre de recherche de l'URL (`query param`) comme suit :
`/campagnes/creation?source=#identifiant de la campagne source#`.


## :rainbow: Remarques

L'ensemble des attributs est copié dont le propriétaire et l'organisation.
Seul le nom de la campagne est préfixé par "Copie de " afin de montrer au créateur de la campagne qu'il vient de dupliquer une campagne.


On en profite pour faire un coup de propre sur le formulaire de création de campagne et les tests associés.


## :100: Pour tester

Se connecter à Pix Orga.
Saisir l'URL de création de campagne en spécifiant la campagne à dupliquer : `/campagnes/creation?source=#identifiant de la campagne source#`.
Vérifier que le formulaire affiché est pré-rempli.
Tester pour des campagnes sources de différents types et différents paramètres (avec et sans libellé d'identifiant externe, campagne d'évaluation, envoi multiple, etc).

VincentHardouin committed:
refactor(orga): use getter instead of useless tracked properties

Co-authored-by: Guillaume Lagorce <guillaume.lagorce@pix.fr>

VincentHardouin committed:
feat(orga): allow create campaign from another

VincentHardouin committed:
refactor(orga): move type business rule in model

Co-authored-by: Guillaume Lagorce <guillaume.lagorce@pix.fr>

HEYGUL committed:
refactor (orga): simplify campaign value updates

Co-authored-by: <vincent.hardouin@pix.fr>

HEYGUL committed:
chore (orga): remove unused tracked props

HEYGUL committed:
chore (orga): remove useless property

Co-authored-by: <vincent.hardouin@pix.fr>

VincentHardouin committed:
refactor(orga): remove useless tracked properties ownerId

VincentHardouin committed:
refactor(orga): use store object instead of ember object in tests

Co-authored-by: Guillaume Lagorce <guillaume.lagorce@pix.fr>

VincentHardouin committed:
feat(orga): use campaign attributes in create-form

Co-authored-by: Guillaume Lagorce <guillaume.lagorce@pix.fr>

VincentHardouin committed:
refactor(orga): remove useless setProperties in new controller

Co-authored-by: Guillaume Lagorce <guillaume.lagorce@pix.fr>

pix-bot-github commented:
Une fois les applications déployées, elles seront accessibles via les liens suivants :
  * [App (.fr)](https://app-pr6586.review.pix.fr)
  * [App (.org)](https://app-pr6586.review.pix.org)
  * [Orga (.fr)](https://orga-pr6586.review.pix.fr)
  * [Orga (.org)](https://orga-pr6586.review.pix.org)
  * [Certif (.fr)](https://certif-pr6586.review.pix.fr)
  * [Certif (.org)](https://certif-pr6586.review.pix.org)
  * [Admin](https://admin-pr6586.review.pix.fr)
  * [API](https://api-pr6586.review.pix.fr/api/)
  * [Pix1D](https://1d-pr6586.review.pix.fr)

Les variables d'environnement seront accessibles via les liens suivants :
  * [scalingo front](https://dashboard.scalingo.com/apps/osc-fr1/pix-front-review-pr6586/environment)
  * [scalingo api](https://dashboard.scalingo.com/apps/osc-fr1/pix-api-review-pr6586/environment)

VincentHardouin committed:
feat(orga): add duplicate campaign e2e test

Co-authored-by: Guillaume Lagorce <guillaume.lagorce@pix.fr>

VincentHardouin committed:
feat(api): add seed for prescription team

Co-authored-by: Guillaume Lagorce <guillaume.lagorce@pix.fr>

HEYGUL committed:
feat (orga): handle case where searching for source campaign throws an error

when searching for source campaign by id, two main reasons can lead to
an error:
- campaign does not exist
- campaign exists but current user does not have sufficient privileges
  to access it

in every case we are just ignoring the error and display the creation
form in the empty state like any campaign creation

Alexandre-Monney commented on orga/app/components/campaign/create-form.hbs:
```diff
@@ -29,7 +29,7 @@
       @options={{this.campaignOwnerOptions}}
       @onChange={{this.onChangeCampaignOwner}}
-      @value={{this.ownerId}}
+      @value="{{@campaign.ownerId}}"
```
A modifier en this.campaign

Alexandre-Monney reviewed (approved):
Revue tech en mob 👍 🐱

Alexandre-Monney commented:
Revue Func Ok 👍
